dags/portal-data-to-snowflake-dag.py

```diff
@@ -0,0 +1,132 @@
+        for col in portal_df.columns
+    ]
+    update_set_str = ",".join(update_set)
+    col_str = ",".join(f'"{col}"' for col in portal_df.columns)
+    to_insert_str = ",".join([f'"{temp_table}"."{col}"' for col in portal_df.columns])
+    merge_sql = f"""
+    MERGE INTO {portal_name} USING {temp_table}
+        ON {portal_name}.{id_col} = {temp_table}.{id_col}
+        when matched then
+            update set {update_set_str}
+        when not matched then
+        insert
+        ({col_str}) values({to_insert_str});
+    """
+    return merge_sql
+
+
+dag_config = {
+    "schedule_interval": "0 0 * * 1",
+    "start_date": datetime(2024, 1, 10),
+    "catchup": False,
+    "default_args": {
+        "retries": 1,
+    },
+    "tags": ["snowflake"],
+    "params": dag_params,
+}
+
+
+@dag(**dag_config)
+def portal_data_to_snowflake():
+    @task
+    def get_portal_data_from_synapse(**context):
+        syn_hook = SynapseHook(context["params"]["synapse_conn_id"])
+        for portal_name, info in portal_dict.items():
+            synapse_id = info["synapse_id"]
+            # Allow for Synapse IDs with version numbers
+            ent = syn_hook.client.get(synapse_id.split(".")[0])
+            if isinstance(ent, synapseclient.EntityViewSchema):
+                portal = syn_hook.client.tableQuery(f"select * from {synapse_id}")
+                portal_df = portal.asDataFrame()
+                portal_df.reset_index(inplace=True, drop="index")
+            elif isinstance(ent, synapseclient.File):
+                portal_df = pd.read_csv(ent.path)
+            else:
+                raise TypeError(f"Unsupported entity type: {type(ent)}")
+            # "grant" and "group" are reserved key words
+            portal_df.rename(
+                columns={"grant": "grants", "group": "groups"}, inplace=True
+            )
+            # Standardize column names for Snowflake tables
+            portal_df.columns = [col.upper() for col in portal_df.columns]
+            portal_dict[portal_name]["data"] = portal_df
+        return portal_dict
+
+    @task
+    def upsert_portal_data_to_snowflake(portal_data, **context):
+        snow_hook = SnowflakeHook(context["params"]["snowflake_conn_id"])
+        for portal_name, info in portal_data.items():
+            # Create temporary table so we can upsert
+            temp_table = f"{portal_name}_TEMP"
+            write_pandas(
+                snow_hook.get_conn(),
+                info["data"],
+                temp_table,
+                auto_create_table=True,
+                table_type="transient",
+                overwrite=True,
+                quote_identifiers=False,
+            )
+            merge_sql = prepare_merge_sql(
+                portal_name=portal_name,
+                portal_df=info["data"],
+                temp_table=temp_table,
+                id_col=info["id_col"],
+            )
+            # TODO account for schema changes
```

Comment:
Does anything need to be done about this at the moment, or is there a follow up jira to address this?

Reply:
This is a note left by @thomasyu888 from the original script that he was running manually to perform this work. I left this in the DAG to be sure it was addressed if needed in the future. I don't know if there is a Jira ticket.

Reply:
@BWMac I think this script will currently fail if there are column changes in any of the tables.  

I forgot but thats actually the reason why I switched to just overwrite the tables in snowflake instead of an upsert.  Lets create a ticket to either switch to overwriting the table or supporting schema changes.

Reply:
@thomasyu888 Should this new ticket be completed as part of this PR before merging?

Reply:
Probably afterword? But we would need to be clear that the created ticket would be a blocker for any future schema changes.

Reply:
Per slack DMs with @thomasyu888 I am going to go ahead and do the overwrite method for now since it is simple and then we can create a ticket for supporting upserts + schema changes in the future.

Reply:
I have updated the DAG to use overwriting instead, and I created a [ticket](https://sagebionetworks.jira.com/browse/ORCA-298) for updating this to upsert and allow for schema changes.